base/linalg/triangular.jl

```diff
@@ -283,6 +283,155 @@ function -(A::UnitUpperTriangular)
     UpperTriangular(Anew)
 end
 
+# copy and scale
+function copy!(A::UpperTriangular,B::UpperTriangular)
+    copy!(A.data,B.data)
+end
+function copy!(A::UnitUpperTriangular,B::UnitUpperTriangular)
+    copy!(A.data,B.data)
+end
+function copy!(A::LowerTriangular,B::LowerTriangular)
+    copy!(A.data,B.data)
+end
+function copy!(A::UnitLowerTriangular,B::UnitLowerTriangular)
+    copy!(A.data,B.data)
+end
+function copy(A::Union{UpperTriangular,UnitUpperTriangular,LowerTriangular,UnitLowerTriangular})
+    D = similar(A)
+    copy!(D, A)
+    return D
+end
+
+function scale!(A::UpperTriangular, B::UpperTriangular, c::Number)
+    n = chksquare(B)
+    for j = 1:n
+        for i = 1:j
+            @inbounds A[i,j] = c * B[i,j]
+        end
+    end
+end
+function scale!(A::UpperTriangular, B::UnitUpperTriangular, c::Number)
+    n = chksquare(B)
+    for j = 1:n
+        A[j,j] = c
+        for i = 1:j-1
+            @inbounds A[i,j] = c * B[i,j]
+        end
+    end
+end
+function scale!(A::LowerTriangular, B::LowerTriangular, c::Number)
+    n = chksquare(B)
+    for j = 1:n
+        A[j,j] = c
+        for i = j:n
+            @inbounds A[i,j] = c * B[i,j]
+        end
+    end
+end
+function scale!(A::LowerTriangular, B::UnitLowerTriangular, c::Number)
+    n = chksquare(B)
+    for j = 1:n
+        A[j,j] = c
+        for i = j+1:n
+            @inbounds A[i,j] = c * B[i,j]
+        end
+    end
+end
+
+function scale{R<:Real,C<:Complex}(A::Union{UpperTriangular{R}, LowerTriangular{R}}, c::C)
+    D = similar(A, promote_type(R,C))
+    scale!(D,A,c)
+    return D
+end
+function scale{R<:BigFloat,C<:Complex}(A::UpperTriangular{R}, c::C)
+    D = UpperTriangular(zeros(promote_type(R,C),size(A)...))
+    scale!(D,A,c)
+    return D
+end
+function scale{R<:BigFloat,C<:Complex}(A::LowerTriangular{R}, c::C)
```

Comment:
Because when I use `similar` or `Array` to create a `UpperTriangular{BigFloat}`, for example, the entries below the diagonal are `#undef` rather than `0`, and the equality checks fail.